packages/components/src/components/Header/Header.js

```diff
@@ -11,26 +11,65 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-import React from 'react';
+import React, { Component } from 'react';
 
 import { Link } from 'react-router-dom';
-import { Header as CarbonHeader } from 'carbon-components-react';
-
+import { Button, Header as CarbonHeader } from 'carbon-components-react';
+import LogoutIcon from '@carbon/icons-react/lib/logout/20';
 import './Header.scss';
+import { shouldDisplayLogout } from '../../../../../src/api';
 
 import tektonBanner from '../../images/tekton-horizontal-color.png';
 
-export default function Header({ children }) {
-  return (
-    <CarbonHeader aria-label="Tekton Dashboard" className="header">
-      <Link to="/">
-        <img
-          alt="Tekton logo and title"
-          className="banner"
-          src={tektonBanner}
-        />
-      </Link>
-      {children && <div className="header-children">{children}</div>}
-    </CarbonHeader>
-  );
+export class Header extends Component {
+  state = {
+    showLogout: false
+  };
+
+  componentDidMount() {
+    this.determineOpenShift();
+  }
+
+  async determineOpenShift() {
+    try {
+      const detectedOpenShift = await shouldDisplayLogout();
+      this.setState({
+        showLogout: detectedOpenShift
+      });
+    } catch (error) {} // eslint-disable-line
+  }
+
+  handleLogoutClick() { // eslint-disable-line
+    window.location.href = '../../oauth/sign_out';
+  }
+
+  render() {
+    return (
+      <CarbonHeader aria-label="Tekton Dashboard" className="header">
+        <Link to="/">
+          <img
+            alt="Tekton logo and title"
+            className="banner"
+            src={tektonBanner}
+          />
+        </Link>
+        {this.state.showLogout && (
+          <Button
+            data-testid="logout-btn"
+            kind="ghost"
+            className="logout-btn"
+            renderIcon={LogoutIcon}
+            onClick={this.handleLogoutClick}
+          >
+            Log out
+          </Button>
+        )}
+        {this.props.children && (
+          <div className="header-children">{this.props.children}</div>
+        )}
+      </CarbonHeader>
+    );
+  }
 }
+
+export default Header;
```

Comment:
We should make sure any new strings are externalised, docs to come (https://github.com/tektoncd/dashboard/issues/453).

Steps:
- `import { injectIntl } from 'react-intl';`
- `export default injectIntl(Header);`
- instead of the hardcoded 'Log out' string use a message object, e.g.:
  ```
  this.props.intl.formatMessage({
    id: 'dashboard.header.logOut',
    defaultMessage: 'Log out'
  })
  ```
- and add it to `src/constants.js` (we're working on tooling to avoid this manual step)

Reply:
I'll do this, thanks